core/src/main/java/org/apache/gravitino/job/JobExecutorFactory.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.job;
+
+import com.google.common.base.Preconditions;
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.gravitino.Config;
+import org.apache.gravitino.Configs;
+import org.apache.gravitino.connector.job.JobExecutor;
+
+public class JobExecutorFactory {
+
+  private static final String JOB_EXECUTOR_CONF_PREFIX = "gravitino.executor.";
+
+  private static final String JOB_EXECUTOR_CLASS_SUFFIX = ".class";
+
+  private JobExecutorFactory() {
+    // Private constructor to prevent instantiation
+  }
+
+  public static JobExecutor create(Config config) {
```

Comment:
Should we use an isolated classloader for the job executor to avoid class conflict with Gravitino server?

Reply:
JobExecutor typically is an executor connecting to the remote job scheduler like Airflow, or local executor (isolated processor), which doesn't have classloader problem. So, we don't have to use isolated classloader.